dkg/messages.go

```diff
@@ -113,6 +116,11 @@ func (signedMsg *SignedMessage) GetRoot() ([]byte, error) {
 	return signedMsg.Message.GetRoot()
 }
 
+type InitOrReshare struct {
```

Comment:
I'd separate that into 2 distinct messages ..

Reply:
Combining these two into one is because the `protocol` interface has one method `Start(initOrReshare InitOrReshare) error` which accepts one message. It's confusing also to pass two arguments where one has to be `nil`. I would think a more logical way is to have pass `Init` or `Reshare` in `New` and `Start` will not need to take the input. Think about `Init` / `Reshare` triggers the creation of the protocol instance while `Start` kicks off the protocol.

Reply:
Sure, suggest a change and lets do it

Reply:
Made the changes in https://github.com/bloxapp/ssv-spec/pull/82/commits/4d0e75bd6f3648d4ecd6ae92fea16064e572a818 and https://github.com/bloxapp/ssv-spec/pull/82/commits/30d3461b1202c1ea345831bccf3c148b6f1a8260